### ISSUE TITLE
cast.ToInt64E("") returns error while cast.ToInt64E(nil) doesn't

### ISSUE THREAD

**Issue by andig** [spf13/cast#212]
Wondering if empty string shouldn't be treated similar to nil?

**Comment by prochac** [comment]
nil/NULL points is 0x00 memory, "" doesn't exactly feel like `int64(0)`

What's the point of `E` when anything returns 0, not error?

**Comment by andig** [comment]
> nil/NULL points to 0x00 memory, "" doesn't exactly feel like int64(0)

Agreed that's probably up to opinion. From user perspective "I don't have a value" seems equivalent to "I don't have a value in that string".

>What's the point of E when anything returns 0, not error?

It doesn't. "foo" ist still an error afaikt.
